packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/drp/dedicatedCloud-datacenter-drp.routing.js

```diff
@@ -23,16 +23,86 @@ export default /* @ngInject */ ($stateProvider) => {
         ),
       disableForUS: /* @ngInject */ ($q, isDrpAvailable) =>
         !isDrpAvailable ? $q.reject() : $q.when(),
+      setDisableSuccessAlertPreference: /* @ngInject */ (dedicatedCloudDrp) => (
+        pccId,
+        value,
+      ) => dedicatedCloudDrp.setDisableSuccessAlertPreference(pccId, value),
       selectedDrpType: /* @ngInject */ ($transition$) => ({
         id: get($transition$.params().selectedDrpType, 'id', null),
       }),
+      setupConfiguration: /* @ngInject */ (
+        $q,
+        $translate,
+        $timeout,
+        $window,
+        dedicatedCloudDrp,
+        displayErrorMessage,
+        displayInfoMessage,
+        displaySuccessMessage,
+        goToPccDashboard,
+        setDisableSuccessAlertPreference,
+        storeZertoOptionOrderInUserPref,
+        DEDICATED_CLOUD_CONSTANTS,
+      ) => (drpInformations) =>
+        dedicatedCloudDrp
+          .enableDrp(
+            drpInformations,
+            drpInformations.primaryPcc.generation !==
+              DEDICATED_CLOUD_CONSTANTS.pccNewGeneration,
+          )
+          .then((enableDrp) => {
+            const storeOptionPromise =
+              enableDrp.url !== undefined
+                ? storeZertoOptionOrderInUserPref(drpInformations, enableDrp)
+                : $q.when();
+
+            const { hasAutoPay } = enableDrp;
+            if (!hasAutoPay) {
+              $window.open(enableDrp.url, '_blank');
+            }
+
+            return $q
+              .all([
+                storeOptionPromise,
+                setDisableSuccessAlertPreference(
+                  drpInformations.primaryPcc.serviceName,
+                  false,
+                ),
+              ])
+              .then(() => goToPccDashboard(true))
+              .then(() => {
+                if (!hasAutoPay) {
```

Comment:
Why is there the `autoPay` check twice ? I'm having trouble understanding here, we open a new window but stil wait for a promise to display a feedback. Could we do both at the same time ?